modules/xfem/src/base/XFEMSquareCut.C

```diff
@@ -0,0 +1,67 @@
+/****************************************************************/
+/*               DO NOT MODIFY THIS HEADER                      */
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*           (c) 2010 Battelle Energy Alliance, LLC             */
+/*                   ALL RIGHTS RESERVED                        */
+/*                                                              */
+/*          Prepared by Battelle Energy Alliance, LLC           */
+/*            Under Contract No. DE-AC07-05ID14517              */
+/*            With the U. S. Department of Energy               */
+/*                                                              */
+/*            See COPYRIGHT for full restrictions               */
+/****************************************************************/
+
+#include "XFEMSquareCut.h"
+
+#include "MooseError.h"
+#include "XFEMFuncs.h"
+#include "EFAFuncs.h"
+
+XFEMSquareCut::XFEMSquareCut(std::vector<Real> square_nodes):
```

Comment:
This could just take a vector of Points. Then you do not have to convert stuff around, plus the code would be more expressive.